src/app/dao/tests/JoinerDAO.spec.ts

```diff
@@ -1,17 +1,446 @@
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should forbid creating a joiner on behalf of another user', async() => {
+            // Given two users, including a malicious verified one, and an existing part
+            const createdPart: CreatedPart = await createPart({ createJoiner: false, signOut: true });
+            await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            // When the malicious user creates the corresponding joiner on behalf of the regular user
+            const result: Promise<void> = joinerDAO.set(createdPart.partId,
+                                                        { ...JoinerMocks.INITIAL, creator: createdPart.creator });
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should forbid a to use a fake name in the list of candidates', async() => {
+            // Given a part and joiner, and a verified user
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const candidate: MinimalUser = await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            // When the user adds themself to the list of candidates with a fake name
+            const result: Promise<void> = joinerDAO.addCandidate(partId, { id: candidate.id, name: 'blibli' });
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should forbid a to add anyone else to the candidates', async() => {
+            // Given a part and joiner, and two verified users
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const attackedCandidate: MinimalUser = await createConnectedUser(CANDIDATE_EMAIL, CANDIDATE_NAME);
+            await signOut();
+            await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            // When the user adds someone else to the list of candidates
+            const result: Promise<void> = joinerDAO.addCandidate(partId, attackedCandidate);
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should allow to add themself to the candidates', async() => {
+            // Given a part and joiner, and a verified user
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const candidate: MinimalUser = await createConnectedUser(CANDIDATE_EMAIL, CANDIDATE_NAME);
+
+            // When the user adds themself to the list of candidates
+            const result: Promise<void> = joinerDAO.addCandidate(partId, candidate);
+
+            // Then it should succeed
+            await expectAsync(result).toBeResolvedTo();
+        });
+        it('should forbid to remove someone else from the candidates', async() => {
```

Comment:
ça c'est à proprement parler pas valid (on est dans la catégorie(describe) "vérified user" et l'interdiction de kicker quelqu'un de la room est possible pour les uns impossible pour les autres, donc p-e qu'hygièniquement ce test doit pas exister, mais une catégorie(describe) "for creator" et "for candidate" devrais avoir chacun leur test ! Non ?

Reply:
tu veux dire dupliquer ce test dans for creator et for candidate ? Ça me semble tout à fait adapté, j'attends confirmation avant de le faire. Mais kicker est interdit pour quiconque non ?

Reply:
Le créateur peut kicker un adversaire timeouté non ? si il échoue à s'enlever lui même ?